lib/ldp.js

```diff
@@ -156,42 +154,40 @@ class LDP {
       return callback(error(500, "Can't parse container"))
     }
 
-    async.waterfall(
-      [
-        // add container stats
-        function (next) {
-          ldpContainer.addContainerStats(ldp, reqUri, filename, resourceGraph, next)
-        },
-        // reading directory
-        function (next) {
-          ldpContainer.readdir(filename, next)
-        },
-        // Iterate through all the files
-        function (files, next) {
-          async.each(
-            files,
-            function (file, cb) {
-              let fileUri = url.resolve(reqUri, encodeURIComponent(file))
-              ldpContainer.addFile(ldp, resourceGraph, reqUri, fileUri, uri,
-                filename, file, cb)
-            },
-            next)
-        }
-      ],
-      function (err, data) {
+    // add container stats
+    new Promise((resolve, reject) =>
+      ldpContainer.addContainerStats(ldp, reqUri, filename, resourceGraph,
+        err => err ? reject(err) : resolve())
+    )
+    // read directory
+    .then(() => new Promise((resolve, reject) =>
+      ldpContainer.readdir(filename,
+        (err, files) => err ? reject(err) : resolve(files))
```

Comment:
We could do that—and there are a couple of libraries that expose such a function (starting with Node 8 even in core). I've used that in other projects.

I was hesitant to do this here and in other places of the Solid code, because callbacks are sometimes used inconsistently (they don't always have an error argument). So I could start using it here, but then it would perhaps be good to keep an eye on it in other places of the code as well. And eventually, we should migrate to an `await`/`async` style, I suppose.

So, what do you think, should we start the promisify practice here?
Perhaps we should consider switching to `async`/`await` (with build step) for node-solid-server v5? Or already start it here?